src/main/serverHelper.ts

```diff
@@ -0,0 +1,71 @@
+import http from 'http';
+import Logging from '@/utils/logging';
+
+const console = Logging.server;
+
+export function basicAuth(expectedUser: string, expectedPassword: string, authString: string): boolean {
+  if (!authString) {
+    console.log('Auth failure: no username+password given');
+
+    return false;
+  }
+  const m = /^Basic\s+(.*)/.exec(authString);
+
+  if (!m) {
+    console.log('Auth failure: only Basic auth is supported');
+
+    return false;
+  }
+  const [user, ...passwordParts] = base64Decode(m[1]).split(':');
+  const password = passwordParts.join(':');
+
+  if (user !== expectedUser || password !== expectedPassword) {
+    console.log(`Auth failure: user/password validation failure for attempted login of user ${ user }`);
+
+    return false;
+  }
+
+  return true;
+}
+
+function base64Decode(value: string): string {
+  return Buffer.from(value, 'base64').toString('utf-8');
+}
+
+export async function getRequestBody(request: http.IncomingMessage, maxPayloadSize: number): Promise<[string, string]> {
+  const chunks: Buffer[] = [];
+  let error = '';
+  let dataSize = 0;
+
+  // Read in the request body
+  for await (const chunk of request) {
+    dataSize += chunk.toString().length;
```

Comment:
Don't eagerly convert chunks to string just to get the length.  We can probably get away with defining `maxPayloadSize` to be in bytes instead of characeters.

Reply:
Sure. The limit is big enough that counting bytes instead of characters shouldn't penalize non-ASCII users.